edb/edgeql/parser/grammar/convert.py

```diff
@@ -0,0 +1,418 @@
+#
+# This source file is part of the EdgeDB open source project.
+#
+# Copyright 2008-present MagicStack Inc. and the EdgeDB authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import parsing
+import pathlib
+from typing import Dict, List, Sequence, Tuple, Callable
+
+
+class EBNF_Item:
+    pass
```

Comment:
I'd recommend using either `dataclass` or else our own `ast` mechanism. (Probably `dataclass`).

You get to skip writing `__init__`, the attributes and their types are spelled out more cleanly, and you automatically get string formatting.